src/main/java/edu/harvard/iq/dataverse/settings/SettingsServiceBean.java

```diff
@@ -486,9 +484,22 @@ public Long getValueForKeyAsLong(Key key){
             logger.log(Level.WARNING, "Incorrect setting.  Could not convert \"{0}\" from setting {1} to long.", new Object[]{val, key.toString()});
             return null;
         }
-        
+
+    }
+    
+    public Integer getValueForKeyAsInt(Key key){
+        Long value = getValueForKeyAsLong(key);
+        if (value == null) {
+            return null;
```

Comment:
I understand that it's necessary to return null otherwise we would need big refactor? Otherwise I would return Optional or default value.

Reply:
The problem with default value is picking the right one default value :)
If we for example choose 0 then we would have broke for example setting: SearchHighlightFragmentSize - currently null means do not override default solr setting but 0 is also correct value that means set solr param to 0 (functionality disabled)
It is not that it would be big refactor. But rather it will require to carefully pick the right one value.
Optional will require more changes in code.

Reply:
Yeah Optional seems more reasonable, but let's just leave it for refactoring tasks.